kernel/src/snapshot.rs

```diff
@@ -546,4 +553,101 @@ mod tests {
             Some(3)
         );
     }
+
+    #[test]
+    fn test_snapshot_version_0_with_checkpoint_at_version_1() {
+        let path =
+            std::fs::canonicalize(PathBuf::from("./tests/data/app-txn-checkpoint/")).unwrap();
+        let url = url::Url::from_directory_path(path).unwrap();
+
+        let engine = SyncEngine::new();
+
+        // First, let's verify the content of the _last_checkpoint file
+        let fs_client = engine.get_file_system_client();
+        let log_url = LogPath::new(&url).child("_delta_log/").unwrap();
+        println!("Log root: {}", log_url);
+        let last_checkpoint = read_last_checkpoint(fs_client.as_ref(), &log_url).unwrap();
+
+        assert!(
+            last_checkpoint.is_some(),
+            "_last_checkpoint file should exist"
+        );
+        let checkpoint_meta = last_checkpoint.unwrap();
+        println!("Checkpoint metadata: {:#?}", checkpoint_meta);
+        assert_eq!(
+            checkpoint_meta.version, 1,
+            "Last checkpoint should be at version 1"
+        );
+        assert_eq!(checkpoint_meta.size, 8, "Checkpoint size should be 8");
+        // assert_eq!(
+        //     checkpoint_meta.size_in_bytes,
+        //     Some(21857),
+        //     "Checkpoint size in bytes should be 21857"
+        // );
```

Comment:
Why commented out, out of curiosity? Is the byte count wrong and needs to be updated?

Reply:
I'm getting a `None` result for `checkpoint_meta.size_in_bytes`; hence, the test is failing. I commented the code so that I first focus on the snapshot versioning bug! I'm looking into this one, too, now.

Reply:
The root cause of the issue is a mismatch between the casing conventions used in the JSON data and the struct definition. The struct is expecting camelCase (e.g., `sizeInBytes`) due to the `#[serde(rename_all = "camelCase")]` attribute, but the actual JSON data is using snake_case (`size_in_bytes`).

Your short-term fix of adding an alias is a good temporary solution:

```rust
#[serde(alias = "size_in_bytes")]
pub(crate) size_in_bytes: Option<i64>,
```

However, for a long-term solution, we need to address the inconsistency between the Delta protocol specification and the implementation in delta-rs (I built my test data from `delta-rs,` assuming that the other test data was also created using the same, including the last_checkpoint file).

According to the [Delta protocol documentation](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#last-checkpoint-file-schema), the last checkpoint file schema should indeed use camelCase for field names. The fact that delta-rs is writing the metadata in snake_case suggests a deviation from the protocol specification.

The long-term solution should involve:

1. Align the delta-rs implementation with the Delta protocol specification by ensuring the fields in the last checkpoint file are written using camelCase field names.
2. Updating the `CheckpointMetadata` struct to expect camelCase without needing aliases.
3. If any backward compatibility is required, consider implementing a more robust deserialization to handle camelCase and snake_case variants.

It would be worth investigating why delta-rs is writing the metadata in snake_case contrary to the protocol specification. 

What are your thoughts @scovich ?

Reply:
Created a new issue to move this conversation https://github.com/delta-incubator/delta-kernel-rs/issues/326 . Would love to know your thoughts on the issue link.